input/posts/2020-09-02-migrating-wcf-to-grpc.md

```diff
@@ -0,0 +1,507 @@
+
+# Migrate a WCF request-reply service to a gRPC unary RPC
+
+_This chapter is based on the ["Migrate a WCF request-reply service to a gRPC unary RPC"](https://docs.microsoft.com/en-us/dotnet/architecture/grpc-for-wcf-developers/migrate-request-reply) section in Microsoft's documentation. Please read that guide for a better understanding of the original WCF service._
+
+## Migrating the Data Contracts and Service Contracts
+
+The WCF service is a simple request-reply Portfolio service that let's you download either a single portfolio or all portfolios for a given trader. The service and data contracts are defined as following:
+
+```csharp
+[ServiceContract]
+public interface IPortfolioService
+{
+    [OperationContract]
+    Task<Portfolio> Get(Guid traderId, int portfolioId);
+
+    [OperationContract]
+    Task<List<Portfolio>> GetAll(Guid traderId);
+}
+```
+```csharp
+[DataContract]
+public class Portfolio
+{
+    [DataMember]
+    public int Id { get; set; }
+
+    [DataMember]
+    public Guid TraderId { get; set; }
+
+    [DataMember]
+    public List<PortfolioItem> Items { get; set; }
+}
+
+[DataContract]
+public class PortfolioItem
+{
+    [DataMember]
+    public int Id { get; set; }
+
+    [DataMember]
+    public int ShareId { get; set; }
+
+    [DataMember]
+    public int Holding { get; set; }
+
+    [DataMember]
+    public decimal Cost { get; set; }
+}
+```
+
+Before migrating the Data Contracts and Service Contract to gRPC, I'd recommend creating a new class library for the shared contracts. These contracts can then be easily shared between the Server and the Client either via a project reference or package reference, depending on the structure of you WCF solution. Once we have created the class library we'll just copy over the source files and start migrating to gRPC.
+
+The Data Contracts, unlike when migrating to gRPC using Google.Protobuf, will require minimal changes. The only thing we need to do is to define the `Order` property in the `DataMember` attribute. This is equivalent of defining field numbers when creating messages in `.proto` format. These field numbers are used to identify your fields in the message binary format, and should not be changed once your message type is in use. For more information about field numbers, see the [Protocol Buffers language guide](https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers).
```

Comment:
Consider "Unlike when migrating to gRPC using Google.Protobuf, the Data Contracts will require minimal changes."

Reply:
"This is the equivalent of defining..." (add "the")